vmdb/app/controllers/miq_request_controller.rb

```diff
@@ -426,112 +426,58 @@ def get_request_tab_type
     end
 
-    # Add time condition
-    cond_hash = Hash.new
-    cond_hash["AFTER"] = {"value"=>"#{opts[:time_period].to_i} Days Ago","field"=>"MiqRequest-created_on"}
-    cond.push(cond_hash)
+    cond.push("or" => request_types_for_model.collect { |k| {"=" => {"value" => k.to_s, "field" => "MiqRequest-resource_type"}} })
 
-    case @layout
-    when "miq_request_ae"
-      req_typ = :AutomationRequest
-    when "miq_request_host"
-      req_typ = :Host
-    else
-      req_typ = :Vm
-    end
-    request_types = MiqRequest::MODEL_REQUEST_TYPES[req_typ]
-    request_types.each_with_index do |typ,i|
-      typ.each do |k|
-        if k.class == Symbol
-          if i == 0
-            @or_hash = Hash.new   # need this  incase there are more than one type
-            @or_hash["or"] = Array.new
-            cond_hash = Hash.new
-            cond_hash["="] = {"value"=>k.to_s,"field"=>"MiqRequest-resource_type"}
-            @or_hash["or"].push(cond_hash)
-          elsif i == request_types.length-1
-            cond_hash = Hash.new
-            cond_hash["="] = {"value"=>k.to_s,"field"=>"MiqRequest-resource_type"}
-            @or_hash["or"].push(cond_hash)
-          else
-            cond_hash = Hash.new
-            cond_hash["="] = {"value"=>k.to_s,"field"=>"MiqRequest-resource_type"}
-            @or_hash["or"].push(cond_hash)
-          end
-        end
-      end
+    if opts[:type_choice] && opts[:type_choice] != "all"  # Add request_type filter, if selected
+      cond.push("=" => {"value" => opts[:type_choice], "field" => "MiqRequest-request_type"})
     end
-    cond.push(@or_hash)
 
-    if opts[:type_choice] && opts[:type_choice] != "all"  # Add request_type filter, if selected
-      cond_hash = Hash.new
-      cond_hash["="] = {"value"=>opts[:type_choice],"field"=>"MiqRequest-request_type"}
-      cond.push(cond_hash)
+    if opts[:reason_text].present?
+      cond.push(prov_condition_reason_text_expression_key => {"value" => prov_condition_reason_text_sanitized, "field" => "MiqRequest-reason"})
     end
 
-    if opts[:reason_text]  && opts[:reason_text] != ""
-      cond_hash = Hash.new
-      if opts[:reason_text].starts_with?("*") && opts[:reason_text].ends_with?("*")   # Replace beginning/ending * chars with % for SQL
-        hash_key = "INCLUDES"
-        reason_text = opts[:reason_text][1..-2]
-      elsif opts[:reason_text].starts_with?("*")
-        hash_key = "ENDS WITH"
-        reason_text = opts[:reason_text][1..-1]
-      elsif opts[:reason_text].ends_with?("*")
-        hash_key = "STARTS WITH"
-        reason_text = opts[:reason_text][0..-2]
-      else
-        hash_key = "INCLUDES"
-        reason_text = opts[:reason_text]
-      end
+    MiqExpression.new("and" => cond)
+  end
 
-      cond_hash["#{hash_key}"] = {"value"=>reason_text,"field"=>"MiqRequest-reason"}
-      cond.push(cond_hash)
-    end
+  def request_types_for_model
```

Comment:
Hard to tell if they are or aren't already, but these new methods should probably be private.